### PR TITLE
feature: add support for querying simd16 eu per dss

### PR DESCRIPTION
**Pull request by JablonskiMateusz**
Related-To: NEO-12012

### CODE REVIEW

**Comment by JablonskiMateusz** [comment]
KMD patch https://lore.kernel.org/intel-xe/20240710220446.2169797-1-lucas.demarchi@intel.com/

**Review by BartoszDunajski — APPROVED** [review]

File: shared/source/os_interface/linux/xe/ioctl_helper_xe.cpp [hunk 1]
```diff
@@ -504,9 +511,10 @@ bool IoctlHelperXe::getTopologyDataAndMap(const HardwareInfo &hwInfo, DrmQueryTo
         topologySize -= itemSize;
         dataPtr = ptrOffset(dataPtr, itemSize);
     }
-
+    receivedEuPerDssInfo |= receivedSimd16EuPerDssInfo;
```
Comment (lucasdemarchi) [inline shared/source/os_interface/linux/xe/ioctl_helper_xe.cpp:514]: why don't you simply add DRM_XE_TOPO_SIMD16_EU_PER_DSS to operate in the same var as DRM_XE_TOPO_SIMD16_EU_PER_DSS? you are not doing anything differently and kernel isn't going to report both. If the kernel reports both in future it actually means it has 2 different types of EUs in that platform, which would be very odd.

File: shared/test/unit_test/os_interface/linux/xe/ioctl_helper_xe_tests.cpp [hunk 2]
```diff
@@ -835,6 +835,61 @@ TEST(IoctlHelperXeTest, givenComputeDssWhenGetTopologyDataAndMapThenResultsAreCo
     }
 }
 
+TEST(IoctlHelperXeTest, givenSimd16EuPerDssAndEuPerDssWhenGetTopologyDataAndMapThenPreferSimd16EuPerDss) {
```
Comment (lucasdemarchi) [inline shared/test/unit_test/os_interface/linux/xe/ioctl_helper_xe_tests.cpp:838]: it's not meant to prefer one vs the other as we don't return both for any platform.  For a non-mock test, an assert that this is true would be ok.

File: third_party/uapi/xe/xe_drm.h [hunk 3]
```diff
@@ -513,6 +520,7 @@ struct drm_xe_query_topology_mask {
 #define DRM_XE_TOPO_DSS_GEOMETRY	(1 << 0)
 #define DRM_XE_TOPO_DSS_COMPUTE		(1 << 1)
 #define DRM_XE_TOPO_EU_PER_DSS		(1 << 2)
```
Comment (lucasdemarchi) [inline third_party/uapi/xe/xe_drm.h:522]: There are other changes in this header that are already available in drm-next. It's probably easier if you just sync the header with drm-next and add on top this change. This way it won't conflict with future updates.
This is what we have in drm-next:
 
```
#define DRM_XE_TOPO_DSS_GEOMETRY        1
#define DRM_XE_TOPO_DSS_COMPUTE         2
#define DRM_XE_TOPO_L3_BANK             3
#define DRM_XE_TOPO_EU_PER_DSS          4
```

Comment (lucasdemarchi) [reply third_party/uapi/xe/xe_drm.h:522]: Also, feel free to commit this once DRM_XE_TOPO_SIMD16_EU_PER_DSS reaches drm-next as per rules in https://docs.kernel.org/gpu/drm-uapi.html:

> The kernel patch can only be merged after all the above requirements are met, but it must be merged to either drm-next or drm-misc-next before the userspace patches land. uAPI always flows from the kernel, doing things the other way round risks divergence of the uAPI definitions and header files.

**Pull request closed without merging**